Purchases/Public/RCPurchases.m

```diff
@@ -661,32 +665,43 @@ - (void)handleRestoreReceiptPostWithInfo:(RCPurchaserInfo *)info
 }
 
 - (void)checkTrialOrIntroductoryPriceEligibility:(NSArray<NSString *> *)productIdentifiers
-                                 completionBlock:(RCReceiveIntroEligibilityBlock)receiveEligibility {
-    [self receiptData:^(NSData * _Nonnull data) {
-        RCLocalReceiptParser *receiptParser = [[RCLocalReceiptParser alloc] init];
-        [receiptParser checkTrialOrIntroductoryPriceEligibilityWithData:data
-                                                     productIdentifiers:productIdentifiers
-                                                             completion:^(NSDictionary<NSString *, NSNumber *> * _Nonnull receivedEligibility,
-                                                                          NSError * _Nullable error) {
-            if (!error) {
-                NSMutableDictionary<NSString *, RCIntroEligibility *> *convertedEligibility = [[NSMutableDictionary alloc] init];
-                
-                for (NSString *key in receivedEligibility.allKeys) {
-                    convertedEligibility[key] = [[RCIntroEligibility alloc] initWithEligibilityStatusCode:receivedEligibility[key]];
-                }
-                
-                CALL_IF_SET_ON_MAIN_THREAD(receiveEligibility, convertedEligibility);
-            } else {
-                NSLog(@"There was an error when trying to parse the receipt locally, details: %@", error.localizedDescription);
-                [self.backend getIntroEligibilityForAppUserID:self.appUserID
-                                                  receiptData:data
-                                           productIdentifiers:productIdentifiers
-                                                   completion:^(NSDictionary<NSString *,RCIntroEligibility *> * _Nonnull result) {
-                    CALL_IF_SET_ON_MAIN_THREAD(receiveEligibility, result);
+                                 completionBlock:(RCReceiveIntroEligibilityBlock)receiveEligibility
+{
+    [self receiptData:^(NSData *data) {
```

Comment:
the logic here should eventually be moved out of purchases and into something like a higher-level intro eligibility manager, which decides whether to use the local or backend version. 
This PR is gigantic enough as it is, though.